### PR TITLE
Austenem/CAT-246 dataset detail workspace updates

### PR DESCRIPTION
**Pull request by austenem**
## Summary

Update menu choices for the workspace buttons in unified views dataset detail pages to allow for both creating and editing workspaces. 

## Design Documentation/Original Tickets

[CAT-246 Jira ticket](https://hms-dbmi.atlassian.net/browse/CAT-246?atlOrigin=eyJpIjoiNDM0YWM4OGE0OGZjNDZkODgyY2I3ZWI0YWVkOTAyOWMiLCJwIjoiaiJ9)

[CAT-507 Jira ticket (design)](https://hms-dbmi.atlassian.net/browse/HMP-484?atlOrigin=eyJpIjoiYTNlMGMxYmQ1Y2M3NDgwNDgwZmI5MjA2NjZiZjQ1OTIiLCJwIjoiaiJ9)

## Testing

Went through both options for both workspace buttons on several dataset pages. 

## Screenshots/Video

https://github.com/user-attachments/assets/afd7baf3-a937-4cb8-a7ab-ff87b3828a9d

## Checklist

- [X] Code follows the project's coding standards
  - [X] Lint checks pass locally
  - [X] New `CHANGELOG-your-feature-name-here.md` is present in the root directory, describing the change(s) in full sentences.
- [ ] Unit tests covering the new feature have been added
- [X] All existing tests pass
- [X] Any relevant documentation in JIRA/Confluence has been updated to reflect the new feature
- [X] Any new functionalities have appropriate analytics functionalities added

### CODE REVIEW

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (john-conroy) [inline context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:16]: Prefer absolute imports especially when traversing multiple levels.

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (john-conroy) [inline context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:64]: Including 'track' in the name suggests its only purpose is tracking. I'd remove track from the fn name here and below.

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (john-conroy) [inline context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:99]: We don't need the `fontSize` prop here?

File: context/app/static/js/components/detailPage/entityHeader/EntityHeaderActionButtons/EntityHeaderActionButtons.tsx
Comment (john-conroy) [inline context/app/static/js/components/detailPage/entityHeader/EntityHeaderActionButtons/EntityHeaderActionButtons.tsx:21]: Also use an absolute import here.

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (john-conroy) [inline context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:81]: Why can't we just pass these as props to a mui `Button` component?

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (john-conroy) [inline context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:32]: `handleOpenMenu` or `handleOpen`?

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (john-conroy) [inline context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:52]: The `uuid` and `hubmap_id` will always be the primary dataset since it comes from the flask data context. Doesn't the helper panel only appear for the current processed dataset in view?

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (john-conroy) [inline context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:28]: Can we use `useRef` instead of `useState`.

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (john-conroy) [inline context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:105]: Again the `uuid` is for the primary dataset.

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (NickAkhmetov) [reply context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:28]: I actually have seen this pattern used in the MUI docs to handle open state, I.e. using this anchor element's presence in the state as a substitute for an "open" boolean. I don't hate it as long as that logic remains intact (I.e open = Boolean(anchorEl))

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (NickAkhmetov) [reply context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:52]: Yes - there is a zustand store that provides the current visible helper panel content.

Also, [uuid] is never nullish due to list wrapper; if we want to fall back to an empty array in this case, [uuid].filter(Boolean) should work.

**Review by NickAkhmetov — CHANGES_REQUESTED** [review]
The `uuid` mismatches are the main thing that concerns me since that's the same issue that led to the failing downloads regression.

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (austenem) [reply context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:99]: Adding a `fontSize` prop of any value to this icon shrinks it, not sure why - but the default size matches designs.

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (austenem) [reply context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:81]: Because the cases for this component involve very differently styled menu buttons, I opted to let that button be a fully customizable React node prop (screen caps below). This cloned element was an effort to add as many shared properties to that customized button as possible and avoid duplication.

Screenshot 2024-09-06 at 10 30 31 AM
Screenshot 2024-09-06 at 10 30 42 AM

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (austenem) [reply context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:32]: `handleOpen` works!

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (austenem) [reply context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:56]: Got it- updated this to reflect the current processed dataset.

**Review by john-conroy — APPROVED** [review]
Looks good to me. Let's wait on @NickAkhmetov's review before merging.

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (NickAkhmetov) [reply context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:56]: I did suggest the `.filter(Boolean)` approach, but either should work here 😄

Comment (john-conroy) [reply context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:56]: Ignore my comment Austen.

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx
Comment (NickAkhmetov) [reply context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:102]: Icons directly imported from `@mui/material/icons` have a default fontSize of 1.25rem (medium): https://mui.com/material-ui/icons/#size

Icons we import from our icon map have their default font size set to 1rem by the default icon styles, so we end up needing to add the fontSize prop for those cases.

**Review by NickAkhmetov — APPROVED** [review]
🚀

**Comment by austenem** [comment]
Thanks!

**Pull request merged**
